torch/distributed/_tensor/op_schema.py

```diff
@@ -130,6 +130,23 @@ def __str__(self) -> str:
         return f"{tuple_strategies_str}\n{child_strategies_str}"
 
 
+@dataclass
+class RuntimeSchemaInfo:
+    """
+    RuntimeSchemaInfo stores the operator schema related information for runtime (eager)
+    execution. This is mainly used for two ways: 1. to generate hash for args to determine
+    whether to re-run sharding prop or not 2. to determine if we need pytree
+    """
+
+    # This static_argnum records static arg starting index for ops that have non-tensor
+    # args/kwargs which would affect sharding propagation results.
+    # only a few ops need this information, e.g. view, transpose, var.dim, etc.
+    static_argnum: int = -1
```

Comment:
Does this we made the assumption that the first static_argnum args will affect sharding props, and args more than that is not? For a fix, this is fine as long as all DTensor unit test passes. But I am wondering if this might be a hacky assumption here.

Reply:
actually no, the comment says `static_argnum` is the starting index that would affect sharding prop, so the arguments on or after `static_argnum` would influence whether to run the sharding propagation again. I think this is the right way to handle caching uniformly and there's no violation so far in all the ops I skimmed